Amplitude/Amplitude.m

```diff
@@ -401,7 +403,9 @@ - (void)logEvent:(NSString*) eventType withEventProperties:(NSDictionary*) event
             [event setValue:eventType forKey:@"event_type"];
             [event setValue:[NSNumber numberWithLongLong:newId] forKey:@"event_id"];
             [event setValue:[self replaceWithEmptyJSON:eventProperties] forKey:@"event_properties"];
+            (void) SAFE_ARC_AUTORELEASE(eventProperties);
```

Comment:
i think the AUTORELEASES should happen together for better code organization. Also needs to happen if the block returns before this.